### PR TITLE
Add dodsrc and urs cookies files when creating netrc

### PR DESCRIPTION
**Pull request by battistowx**
This PR includes code that will also create a .urs_cookies and .dodsrc files when users persist their credentials to a .netrc file, and will copy the .dodsrc file based on the user operating system. It is based on code from this how-to, but adapted here to use pathlib: https://disc.gsfc.nasa.gov/information/howto?keywords=prerequisite&title=How%20to%20Generate%20Earthdata%20Prerequisite%20Files

I have tested this with poetry on both Linux/Windows, and it works great!

----
📚 Documentation preview 📚: https://earthaccess--492.org.readthedocs.build/en/492/

### CODE REVIEW

File: earthaccess/auth.py [hunk 1]
```diff
@@ -370,6 +372,20 @@ def _persist_user_credentials(self, username: str, password: str) -> bool:
         my_netrc = Netrc(str(netrc_path))
         my_netrc["urs.earthdata.nasa.gov"] = {"login": username, "password": password}
         my_netrc.save()
+        netrc_file = Path.home() / ".netrc"
```
Comment (jhkennedy) [inline earthaccess/auth.py:375]:
```suggestion
```
This is already defined on line 366 as `netrc_path`

File: earthaccess/auth.py [hunk 2]
```diff
@@ -370,6 +372,20 @@ def _persist_user_credentials(self, username: str, password: str) -> bool:
...
+
+        # Create and write .urs_cookies file
+        urs_cookies_file = Path.home() / ".urs_cookies"
+        urs_cookies_file.write_text("")
+
+        # Create and write to .dodsrc file
+        dodsrc_file = Path.home() / ".dodsrc"
+        dodsrc_contents = f"HTTP.COOKIEJAR={urs_cookies_file}\nHTTP.NETRC={netrc_file}"
```
Comment (jhkennedy) [inline earthaccess/auth.py:383]:
```suggestion
        dodsrc_contents = f"HTTP.COOKIEJAR={urs_cookies_file}\nHTTP.NETRC={netrc_path}"
```

**Review by jhkennedy — COMMENTED** [review]
:thinking: looks pretty good; I can see the desire for the `.urs_cookies` and `.dodsrc` (OpenDAP) files. 

As written, code will clobber (overwrite) any existing  `.urs_cookies` and `.dodsrc`  files, which I don't think is desirable, but I'll have to think more about how best to handle existing versions of these files... 

I also left a couple of minor suggestions.

**Comment by battistowx** [comment]
Thank you for the reviewing and the suggestions @jhkennedy ! I added the suggested commits here. In terms of overwriting the existing dodsrc file, for Windows environments, you need to have additional copies of the dodsrc file in each working directory even if it exists in the root user folder. I'm not sure how long the information in the urs_cookies file is valid for, but I have not had any issues when overwriting it repeatedly, and this file is necessary for those who want to use wget/curl, so I have included it here for convenience.

File: earthaccess/auth.py [hunk 3]
```diff
@@ -370,6 +372,19 @@ def _persist_user_credentials(self, username: str, password: str) -> bool:
...
+        urs_cookies_file = Path.home() / ".urs_cookies"
+        urs_cookies_file.write_text("")
+
+        # Create and write to .dodsrc file
+        dodsrc_file = Path.home() / ".dodsrc"
+        dodsrc_contents = f"HTTP.COOKIEJAR={urs_cookies_file}\nHTTP.NETRC={netrc_path}"
+        dodsrc_file.write_text(dodsrc_contents)
+
+        if platform.system() == "Windows":
+            shutil.copy2(dodsrc_file, Path.cwd())
```
Comment (jhkennedy) [inline earthaccess/auth.py:386]:
```suggestion
        urs_cookies_path = Path.home() / ".urs_cookies"
        if not usr_cookies_path.exists():
            urs_cookies_path.write_text("")

        # Create and write to .dodsrc file
        dodsrc_path = Path.home() / ".dodsrc"
        if not dodsrc_path.exists():
            dodsrc_contents = f"HTTP.COOKIEJAR={urs_cookies_file}\nHTTP.NETRC={netrc_path}"
            dodsrc_path.write_text(dodsrc_contents)
        
       if dodsrc_path.exists() and platform.system() == "Windows":
            shutil.copy2(dodsrc_path, Path.cwd())
            
```

Note: I changed `*_file` to `*_path` here to match the style of the rest of the file.

**Review by jhkennedy — COMMENTED** [review]
@battistowx circling back to this. 

I agree overwriting the `.urs_cookies` path is not likely a concern, but I don't think there's a benefit to doing so. 

For `~/.dodsrc`, I think there's actually a decent amount of risk because:
1. there's a lot of configuration that *can* go into the `~/.dodsrc`, e.g.:
   * https://github.com/OPENDAP/libdap4/blob/version-3.20.6/README.dodsrc#L49-L60
   * https://docs.unidata.ucar.edu/nug/current/dap2.html
   * https://docs.unidata.ucar.edu/nug/current/dap4.html
2. Users may want to specify alternate cookie and netrc files, e.g., #480 

How about we only create them if they don't exist?

File: earthaccess/auth.py (hunk 3 above)
Comment (jhkennedy) [reply earthaccess/auth.py:386]: Only creating them if they don't exist is also in line with OpenDAP's NASA Tutorial's `.dodsrc` setup script:
https://github.com/OPENDAP/NASA-tutorials/blob/main/tutorials/setup_dodsrc.sh

Comment (battistowx) [reply earthaccess/auth.py]: This all looks great! Thanks for pointing out the risks, I learn something new about these prerequisite files every day!

**Comment by jhkennedy** [comment]
Woops! bad indent in my suggestion

**Comment by battistowx** [comment]
> Woops! bad indent in my suggestion

No worries! Fixed them.

**Review by jhkennedy — APPROVED** [review]
Looks good to me! :shipit:

**Pull request merged**